Lesson 8/ConsoleFileManager/FileManager.cs

```diff
@@ -0,0 +1,977 @@
+                Console.Write(new string(' ', UserInterface.ScreenWidth));
+            }
+        }
+
+        /// <summary>
+        /// /// <summary>
+        /// Clears the info screen
+        /// </summary>
+        /// </summary>
+        public void ClearRenderedZone()
+        {
+            for (int i = _itemsPerPage + 2; i < Console.WindowHeight - 3; ++i)
+            {
+                Console.SetCursorPosition(1, i);
+                Console.Write(new string(' ', UserInterface.ScreenWidth));
+            }
+        }
+
+        /// <summary>
+        /// Prints the item list
+        /// </summary>
+        /// <param name="page">Page to be printed</param>
+        /// <param name="items">Item list to be printed</param>
+        /// <param name="isSelected">Used to reverse the color of item if it's selected</param>
+        public void PrintList(uint page, IEnumerable<ItemProperties> items, ItemProperties isSelected = null)
+        { 
+            StringBuilder _buffer = new StringBuilder();
+            try
+            {
+                _buffer.Clear();
+
+                Console.CursorVisible = false;
+
+                for (int i = 0; i < PagedList[page].Length; i++)
+                {
+                    ItemProperties item = PagedList[page][i];
+
+                    if (PagedList[page][i] != null)
+                    {
+                        var name = GetPrintableName(item);
+
+                        if (PagedList[page][i].Equals(isSelected) && !ConsoleMode)
+                        {
+                            _buffer.Append(IsAccessible(item)
+                                ? $"\u001b[{i + 2};2H\u001b[7m{name}\u001b[0m"
+                                : $"\u001b[{i + 2};2H\u001b[41m{name}\u001b[0m");
+                        }
+                        else
+                        {
+                            _buffer.Append(IsAccessible(item)
+                                ? $"\u001b[{i + 2};2H\u001b[37m{name}\u001b[0m"
+                                : $"\u001b[{i + 2};2H\u001b[31m{name}\u001b[0m");
+                        }
+                    }
+                    else
+                        break;
+                }
+                Console.Write(_buffer.ToString());
+                DrawSelectedItemInfo();
+
+                Console.CursorVisible = ConsoleMode;
+            }
+            catch
+            {
+                // ignored
+            }
+
+            Console.ResetColor();
+
+            InfoBar();
+        }
+
+        /// <summary>
+        /// Asks for delete confirmation.
+        /// Does nothing if item is not accessible
+        /// </summary>
+        /// <param name="item">Item to delete</param>
+        /// <returns>True if request is confirmed and false if it's not</returns>
+        public bool DeleteRequest(ItemProperties item)
+        {
+            if (!Items.First().Equals(Selected) && IsAccessible(item) && Directory.GetParent(item.ItemPath) != null)
```

Comment:
```suggestion
            if (!Items[0].Equals(Selected) && IsAccessible(item) && Directory.GetParent(item.ItemPath) != null)
```
Это ж очень долго. Это ж список - можно обратиться по номеру элемента за O(1), а не вызывать целый Linq-метод.